### ISSUE TITLE
Broken links and expired domain names

### ISSUE THREAD

**Issue by Hornbydd** [Riverscapes/ConfinementTool#27]
Hi,

As there appears to be no contacts page for the confinement tool I have ended up here to report some significant bugs with the website http://confinement.riverscapes.xyz/

I hope someone reads this and can pass it on to the appropriate people who manage this website. I'm working with Joe on a project but he is away at the moment so unable to contact him directly.

First of all the download link for version 2.2.03 (the latest tool) does not work! Takes you to a 404 page. so if you have been wondering why no one is downloading your tool it is probably that!

All the menu links in the top right of the page (riverscapes, rc tool, rc science...etc) are all broken taking you to a domain name page, so I can't navigate your site. 

A suggestion would be to add a contacts page but as I am unable to navigate the website may be there is one?

Finally I do not know if the download from the failed link is some sort of prepackaged version of the source code on this site (github)?

Hope this gets to someone fast as I would like to use this tool.

Duncan

**Comment by volkcj** [comment]
Hi Duncan,   
Issues for the Confinement tool should be logged in the GitHub Issues for the Riverscapes/Confinement project, so you're in the right place.    We weren't involved in the updates to the confinement.riverscapes.xyz domain but I can see if I can fix the links and/or get access to do so.  

I was able to download the 2.2.03 version directly from GitHub (and via the confinement wiki on GitHub).  Were your issues only on the *xyz domain or were they in GitHub as well?  Please log a separate issue for the xyz domain issues and the GitHub issues and provide the broken link for the GitHub download/page issues.  Thanks!

**Comment by Hornbydd** [comment]
Hi the download link on this page http://confinement.riverscapes.xyz/ is broken, this was the page Joe had sent me too. The  broken menu links are on http://confinement.riverscapes.xyz/ (menu bar at top of the page) not on github. 

I have to admit I'm quite new to the whole github thing but I was able to fork a copy of the tool and download that. As it is a pyt file there is no extra compilation process I can simply view it in Arcmap. At that point I'm hitting errors in the tool. I'm an ArcGIS Desktop developer so I can actually fix some of these and have started to push back my edits to the tools.  As I'm going to be using this tool a lot in the very near future expect more pull requests as I fix and improve the tool.  A simply improvement is that there is absolutely no parameter help for any of the tools, which one creates via the Item Description. SO I have actually started to improve these by copying text from the help pages on http://confinement.riverscapes.xyz/

**Comment by Hornbydd** [comment]
I am visiting the riverscape.xyz site regularly. I want to emphasise how frustrating it is that the menu at the very top of the website is still broken.

![Menu](https://user-images.githubusercontent.com/22773948/44474287-3900ef80-a62a-11e8-9eb8-93cc2edb9c6a.PNG)

Your site is a rich resource of tools often referring to each other and I can often not get to them because I end up at the 404 page. I urge you to fix this as it is clear that the menu should be the main route into all the various tools that this site offers. Ending up at 404 is just harming the credibility of this site.

**Comment by joewheaton** [comment]
Thanks @Hornbydd.... I hear you on 'frustrating'.  I just hadn't a) finished the riverscapes site and b) plumbed the redirect to there yet.

Now I'm working on that, I need some help from @philipbaileynar  or @MattReimer 

### Gandi Domain managing
In Gandi.net (which you both have access to), I've attempted to plumb this to riverscapes.xyz

```
@ 10800 IN SOA ns1.gandi.net. hostmaster.gandi.net. 1536076013 10800 3600 604800 10800
@ 1800 IN A 199.34.228.67
@ 10800 IN MX 10 spool.mail.gandi.net.
@ 10800 IN MX 50 fb.mail.gandi.net.
@ 10800 IN TXT "v=spf1 include:_mailcust.gandi.net ?all"
blog 10800 IN CNAME blogs.vip.gandi.net.
brat 300 IN CNAME riverscapes.github.io.
cname 1800 IN CNAME riverscapes.weebly.com
confinement 300 IN CNAME riverscapes.github.io.
ctt 1800 IN CNAME riverscapes.github.io.
gcd 1800 IN CNAME riverscapes.github.io.
gnat 300 IN CNAME riverscapes.github.io.
imap 10800 IN CNAME access.mail.gandi.net.
pop 10800 IN CNAME access.mail.gandi.net.
rcat 300 IN CNAME riverscapes.github.io.
smtp 10800 IN CNAME relay.mail.gandi.net.
tat 1800 IN CNAME riverscapes.github.io.
webmail 10800 IN CNAME webmail.gandi.net.
wrat 10800 IN CNAME riverscapes.github.io.
www 1800 IN A 199.34.228.67
```
The three lines that matter are:
- `www 1800 IN A 199.34.228.67`,  
- `@ 1800 IN A 199.34.228.67`. 
- `cname 1800 IN CNAME riverscapes.weebly.com`

### Weebly Site Management
What I've got to work with GoDaddy hosted domain was to simply change my site settings to a site I own:
![weebly](https://user-images.githubusercontent.com/9723905/45043177-b153b180-b029-11e8-8509-00714412065a.png)

Then when you hit continue it gives you the IP address you need (in my case `199.34.228.67`) and then you go into Gandi and that's where I get the idea to add : `www 1800 IN A 199.34.228.67`,  and `@ 1800 IN A 199.34.228.67`. 

You'll see that I also attempted adding the cname entry `cname 1800 IN CNAME riverscapes.weebly.com` becuase this is what worked for me in GoDaddy for my joewheaton.org domain (i.e. I have a CNAME entry of `CNAME www joewheaton.weebly.com`) 

So
#### Redirects
Another related issue...  
I tried this to handle Duncan's issue above... that without the www prefix things aren't working.  It seems like it should handle that automatically? 
![301redirects](https://user-images.githubusercontent.com/9723905/45043495-86b62880-b02a-11e8-8765-e34add84f2e5.png)


Also, I added this in Gandi..
![redirect](https://user-images.githubusercontent.com/9723905/45043580-cc72f100-b02a-11e8-8801-a4d2735ec65a.png)

## Bottom Line
End result of all this is when navigating to http://www.riverscapes.xyz/
```
This page isn’t working
www.riverscapes.xyz didn’t send any data.
ERR_EMPTY_RESPONSE
```

**Comment by joewheaton** [comment]
I removed all the 301 redirects in Weebly, and the web forwarding in Gandi. It seems to have no impact either way. I don't think this is necessary it was just a troubleshooting attempt.

**Comment by joewheaton** [comment]
@philipbaileynar wrote me:

> Hey Joe,
> when I put riverscapes.xyz into a browser I get the weebly site
> you might need a hard refresh, cache clear or wait for your DNS to refresh
> http://www.redsome.com/flush-clear-dns-cache-google-chrome-browser/

and take home is I  might need to just be patient...

> y... up to 72hrs
> could be USU or Chrome or further afield

**Comment by joewheaton** [comment]
Okay... @Hornbydd this issue of broken links is now fixed, but it will force me to finally finish the Riverscapes website ;), which I HAVE NOT had time to do yet. A few things to clarify for you and future users:
- This is classic amaeteur open-source stuff - you and all users get what they pay for. That is to say you have paid nothing ;) and so you shouldn't expect the moon. South Fork Research did the best they could under tight time frames and limited budget to make this stuff available and partially documented. However, this IS NOT a polished, professional, well-documented tool. It is a research tool developed for a specific purpose and made available in case others found it easy to adapt or modify for their purposes.
- Your comments appear to reflect some unrealistic expectations:
> As there appears to be no contacts page for the confinement tool I have ended up here to report some significant bugs with the website http://confinement.riverscapes.xyz/
> 
> I hope someone reads this and can pass it on to the appropriate people who manage this website. 
- This seems to imply that a) there are contacts to tap... that does not happen without funding. We do our best to get to these issue boards when we can... b) 'the people who manage this website' are now unemployed. BPA (in their infinite short-sided wisdom) cut the entire program that supported the original development with little warning and @KellyMWhitehead and @volkcj were left high and dry. I currently have no funding myself to support this effort and am just volunteering my time for a cause I believe in... ;)

Sorry to those that expect all tools to be free, work great and all that. This is a clever concept, and a 'research-grade', amateur tool that works pretty well when dealing with exactly the types of data we were dealing with. In the meantime, my hope is we can find some funding and/or a community to improve this.

**Comment by Hornbydd** [comment]
Moon? Hey I was aiming for Mars! I hear you loud and clear, I have a similar issue with the Barrier Assessment Tool I created for the TNC. Any support/tweaks is in my own time. I think fixing the menu is an important step as it is the gateway to the various projects\tools; previously I could not even enter, now I can visit with ease. So thanks for fixing this.
